workspace/sort-packages/src/index.ts

```diff
@@ -9,62 +9,10 @@ export function sequenceGraph (pkgGraph: ProjectsGraph): GraphSequencerResult<st
     keys.map((pkgPath) => [
       pkgPath,
       pkgGraph[pkgPath].dependencies.filter(
-        /* remove cycles of length 1 (ie., package 'a' depends on 'a').  They
-        confuse the graph-sequencer, but can be ignored when ordering packages
-        topologically.
-
-        See the following example where 'b' and 'c' depend on themselves:
-
-          graphSequencer({graph: new Map([
-            ['a', ['b', 'c']],
-            ['b', ['b']],
-            ['c', ['b', 'c']]]
-          ),
-          groups: [['a', 'b', 'c']]})
-
-        returns chunks:
-
-            [['b'],['a'],['c']]
-
-        But both 'b' and 'c' should be executed _before_ 'a', because 'a' depends on
-        them.  It works (and is considered 'safe' if we run:)
-
-          graphSequencer({graph: new Map([
-            ['a', ['b', 'c']],
-            ['b', []],
-            ['c', ['b']]]
-          ), groups: [['a', 'b', 'c']]})
-
-        returning:
-
-            [['b'], ['c'], ['a']]
-
-        */
-        d => d !== pkgPath &&
-        /* remove unused dependencies that we can ignore due to a filter expression.
-
-        Again, the graph sequencer used to behave weirdly in the following edge case:
-
-          graphSequencer({graph: new Map([
-            ['a', ['b', 'c']],
-            ['d', ['a']],
-            ['e', ['a', 'b', 'c']]]
-          ),
-          groups: [['a', 'e', 'e']]})
-
-        returns chunks:
-
-            [['d'],['a'],['e']]
-
-        But we really want 'a' to be executed first.
-        */
-        setOfKeys.has(d))]
+        d => d !== pkgPath && setOfKeys.has(d))]
     )
   )
-  return graphSequencer({
-    graph,
-    groups: [keys],
-  })
+  return graphSequencer(graph, keys)
```

Comment:
why has the API of the function changed?

Reply:
In the older version, the 'groups' parameter more like a concept of priority. In the case of cycles within the graph, if the nodes within a cycle have different execution priorities, it is safe to execute that cycle in the order of priority. For example:

```
graphSequencer({
  graph: new Map([
    ['a', ['b']],
    ['b', ['c']],
    ['c', ['a']],
  ]),
  groups: [['a', 'b', 'c']]
})
```

If we use the same priority, the cycle is unsafe:

```
{
  safe: false,
  chunks: [ [ 'a' ], [ 'c' ], [ 'b' ] ],
  cycles: [ [ 'a', 'b', 'c' ] ]
}
```

However, if we use different priorities:

```
  groups: [['a'], ['b', 'c']]
```

The execution becomes safe:

```
{ safe: true, chunks: [ [ 'a' ], [ 'c' ], [ 'b' ] ], cycles: [] }
```

Currently, pnpm does not have the concept of priorities. All nodes passed as parameters have the same priority when called. Priority is not supported at the moment, but support for subgraphs (currently also requiring all nodes to be passed) is available. If you wish to add priorities in the future, it's a straightforward process - you just need to implement priority checking when identifying cycles.